File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>> a;
        for (int i=0;i<nums.size();i++){
            a.push_back({nums[i],i});
        }
        sort(a.begin(),a.end());
        int i=0,j=a.size()-1;
        while(i<j){
            if(a[i].first+a[j].first==target){
                return {min(a[i].second,a[j].second),
                max(a[j].second,a[i].second)};
            }
            else if(a[i].first+a[j].first<target){
                i++;
            }
            else{
                j--;
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, h: List[int]) -> int:
        l,r=0,len(h)-1
        lm,rm=h[0],h[r]
        ans=0
        while(l<r):
            if(h[l]<=h[r]):
                print(f"{lm},{rm}", end=" ")
                ans+=min(lm,rm)-h[l]
                print(f"loop1->{l},{r},{ans}")
                l+=1
                lm=max(lm,h[l])
            else:
                print(f"{lm},{rm}",end=" ")
                ans+=min(lm,rm)-h[r]
                print(f"loop2->{l},{r},{ans}")
                r-=1
                rm=max(rm,h[r])
        return ans